File: server/src/main/java/com/bk/sbs/exception/ServerErrorCode.java

package com.bk.sbs.exception;

public enum ServerErrorCode {
    SUCCESS(0, "Success"),
    ACCOUNT_REGISTER_FAIL_REASON1(1001, "Account registration failed due to duplicate email"),
    LOGIN_FAIL_REASON1(2001, "Invalid email or password"),
    CHARACTER_CREATE_FAIL_REASON1(3001, "Character creation failed"),
    UNKNOWN_ERROR(Integer.MAX_VALUE, "Unknown error");

    private final int code;
    private final String message;

    ServerErrorCode(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }
}

File: client/unity/Assets/Scripts/ServerErrorCode.cs
//---------------------------------------------------------------------------------
using System.Collections.Generic;

 public enum ServerErrorCode
{
    SUCCESS = 0,
    ACCOUNT_REGISTER_FAIL_REASON1 = 1001,
    LOGIN_FAIL_REASON1 = 2001,
    CHARACTER_CREATE_FAIL_REASON1 = 3001,
    UNKNOWN_ERROR = int.MaxValue,
}

public static class ErrorCodeMapping
{
    public static readonly Dictionary<ServerErrorCode, string> Messages = new Dictionary<ServerErrorCode, string>
    {
        { ServerErrorCode.SUCCESS, "Success" },
        { ServerErrorCode.ACCOUNT_REGISTER_FAIL_REASON1, "Account registration failed due to duplicate email" },
        { ServerErrorCode.LOGIN_FAIL_REASON1, "Invalid email or password" },
        { ServerErrorCode.CHARACTER_CREATE_FAIL_REASON1, "Character creation failed" },
        { ServerErrorCode.UNKNOWN_ERROR, "Unknown error" },
    };
}